python/ray/tune/experiment.py

```diff
@@ -109,6 +109,12 @@ def __init__(self,
 
         config = config or {}
         self._run_identifier = Experiment.register_if_needed(run)
+        self.name = name or self._run_identifier
```

Comment:
name and spec were placed together because that's how you construct an experiment via json

Reply:
huh? dont quite understand why thats important—I just moved it so that i can use `self.name` to create ~`upload_dir`~ `remote_checkpoint_dir` which is passed into the spec.